File: osiris-data-connection/src/io/osiris/data/connection/ConnectionFactory.java
package io.osiris.data.connection;

import io.osiris.data.connection.properties.PropertiesConnection;
import io.osiris.data.connection.xml.XmlConnection;

import java.io.File;
import java.sql.Connection;
import java.sql.SQLException;

public class ConnectionFactory implements ConnectionProvider {

    @Override
    public Connection openConnection() throws ClassNotFoundException, SQLException {
        boolean propertiesFileExist = new File("resources", "connection.properties").exists();

        if (propertiesFileExist) return PropertiesConnection.openConnection();

        return XmlConnection.openConnection();
    }
}

File: osiris-data-connection/src/io/osiris/data/connection/ConnectionFactory.kt
package io.osiris.data.connection

import io.osiris.data.connection.properties.PropertiesConnection
import io.osiris.data.connection.xml.XmlConnection

import java.sql.Connection
import java.sql.SQLException

class ConnectionFactory : ConnectionAdapter {

    @Throws(ClassNotFoundException::class, SQLException::class)
    override fun openConnection(): Connection {
        val url = ClassLoader.getSystemResource("resources/connection.properties")
        return if (url != null) {
            PropertiesConnection.openConnection()
        } else {
            XmlConnection.openConnection()
        }
    }
}
